### PR TITLE
Automatically merge release branch into main

### LINKED ISSUE #6207: Automatically open a PR against `main` when merging changes in the latest release branch
Cf. https://github.com/RasaHQ/rasa-x/issues/2960

### PR DESCRIPTION
**Pull request by katethemate**
closes #6207 

**Proposed changes**:
- Automatically create a PR that merges changes from the release branch into main **and** close older PRs of that sort.

**Status (please check what you already did)**:
- [ ] added some tests for the functionality
- [ ] updated the documentation
- [ ] updated the changelog (please check [changelog](https://github.com/RasaHQ/rasa/tree/main/changelog) for instructions)
- [ ] reformat files using `black` (please check [Readme](https://github.com/RasaHQ/rasa#code-style) for instructions)

### CODE REVIEW

File: .github/workflows/automatic-release-to-main-merger.yml [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+          while read id; do
+            gh pr close $id -d
+          done <pr_ids.txt
+        env:
+            GITHUB_TOKEN: ${{secrets.GITHUB_TOKEN}}
+
+      - name: Notify Slack when closing of outdated merge-PRs failed 💬
+        if: failure()
+        env:
+          SLACK_WEBHOOK: ${{ secrets.K8S_SLACK_INFRASTRUCTURE_SQUAD_MONITORS_WEBHOOK_URL }}
```
Comment (katethemate) [inline .github/workflows/automatic-release-to-main-merger.yml:75]: @mprazz @tczekajlo do I have access to the `${{ secrets.K8S_SLACK_INFRASTRUCTURE_SQUAD_MONITORS_WEBHOOK_URL }}` here as well? 🤔

Comment (mprazz) [reply .github/workflows/automatic-release-to-main-merger.yml:75]: I don't have admin access to the `rasa` repo so I can't confirm the secret is also added here.. @tczekajlo are you able to see it?

Comment (tczekajlo) [reply .github/workflows/automatic-release-to-main-merger.yml:75]: You can use this one `SLACK_INFRASTRUCTURE_SQUAD_MONITORS_WEBHOOK_URL`

Comment (mprazz) [reply .github/workflows/automatic-release-to-main-merger.yml:75]: @katethemate ping me when you've changed the secret and I can review the PR :smile_cat:

Comment (katethemate) [reply .github/workflows/automatic-release-to-main-merger.yml]: thank you @tczekajlo ! @mprazz, it's ready to be reviewed 😊

**Review by mprazz — APPROVED** [review]

**Pull request merged**